v0/util/index.js

```diff
@@ -1793,6 +1793,36 @@ const getDestAuthCacheInstance = destType => {
   return destInf?.authCache || {};
 };
 
+/**
+ * This function removes all those variables which are
+ * empty or undefined or null from all levels of object.
+ * @param {*} obj
+ * @returns payload without empty null or undefined variables
+ */
+const refinePayload = obj => {
+  const refinedPayload = {};
+  Object.keys(obj).forEach(ele => {
+    if (
+      obj[ele] != null &&
+      typeof obj[ele] === "object" &&
+      !Array.isArray(obj[ele])
+    ) {
+      const refinedObject = refinePayload(obj[ele]);
+      if (Object.keys(refinedObject).length !== 0) {
+        refinedPayload[ele] = refinePayload(obj[ele]);
+      }
+    } else if (
+      typeof obj[ele] === "boolean" ||
+      // eslint-disable-next-line no-restricted-globals
```

Comment:
can't we remove this restriction?

Reply:
No, we can't as `isDefinedAndNotNullAndNotEmpty` does not work on boolean and numbers thats why we have to take them into consideration.

Reply:
Done